core/src/main/scala/org/apache/spark/scheduler/cluster/ExecutorInfo.scala

```diff
@@ -72,6 +82,6 @@ class ExecutorInfo(
   override def hashCode(): Int = {
     val state = Seq(executorHost, totalCores, logUrlMap, attributes, resourcesInfo,
       resourceProfileId)
-    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
+    state.filter(_ != null).map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
```

Comment:
This should not need to change, right ? Or unrelated bugfix ?

Reply:
This is semi-related, it's changed so that we can compare executorinfo objects during testing which may have components that are null.